### PR TITLE
feat: completed new issues page ui

### PR DESCRIPTION
**Pull request by TangoBeeAkto**

### CODE REVIEW

File: apps/dashboard/web/polaris_web/web/src/apps/dashboard/pages/dashboard/components/ActivityTracker.jsx [hunk 1]
```diff
@@ -1,124 +1,98 @@
...
+  const latestActivity = generateActivityEvents(issueDetails).reverse()
+  const groupedActivity = groupEventsByDate(latestActivity)
+
+  return (
+      <Box padding={5}>
+          <VerticalStack>
+              {Object.keys(groupedActivity).map((date, dateIndex) => (
+                  <Box key={dateIndex}>
+                      <HorizontalStack gap={5}>
+                        <div style={{ background: '#C9CCCF', width: '2px', marginLeft: '12px', height: '44px' }} />
```
Comment (Ark2307) [inline apps/dashboard/web/polaris_web/web/src/apps/dashboard/pages/dashboard/components/ActivityTracker.jsx:66]: no div

File: apps/dashboard/web/polaris_web/web/src/apps/dashboard/pages/dashboard/components/ActivityTracker.jsx [hunk 2]
```diff
@@ -1,124 +1,98 @@
...
+                        <Text variant="bodySm" color="subdued" style={{ marginBottom: '10px' }}>
+                            {date}
+                        </Text>
+                      </HorizontalStack>
+                      {groupedActivity[date].map((event, eventIndex) => (
+                          <HorizontalStack key={eventIndex} align='space-between'>
+                              <div style={{ display: 'flex', gap: '12px' }}>
```
Comment (Ark2307) [inline apps/dashboard/web/polaris_web/web/src/apps/dashboard/pages/dashboard/components/ActivityTracker.jsx:73]: horizontalStack

File: apps/dashboard/web/polaris_web/web/src/apps/dashboard/pages/dashboard/components/ActivityTracker.jsx [hunk 3]
```diff
@@ -1,124 +1,98 @@
...
-        }
+import './ActivityTracker.css'
+
+function ActivityTracker({ issueDetails }) {
+  const formatDate = (epochTime) => {
+      const date = new Date(epochTime * 1000)
+      const formattedDate = date.toLocaleDateString('en-US', { 
+          year: 'numeric', month: 'short', day: 'numeric' 
+      })
+      const formattedTime = date.toLocaleTimeString('en-US', {
```
Comment (Ark2307) [inline apps/dashboard/web/polaris_web/web/src/apps/dashboard/pages/dashboard/components/ActivityTracker.jsx:11]: func.epochToDateTime{use this}

File: apps/dashboard/web/polaris_web/web/src/apps/dashboard/pages/issues/transform.js [hunk 4]
```diff
@@ -1,82 +1,128 @@
...
+            default:
+                return 'Low'
+        }
+    },
+    
+    getIssueStatus: (type) => {
+        switch (type.toUpperCase()) {
+            case "OPEN":
+                return "Open"
+            case "FIXED":
```
Comment (Ark2307) [inline apps/dashboard/web/polaris_web/web/src/apps/dashboard/pages/issues/transform.js:110]: use func.toSentenceCase, remove this

File: apps/dashboard/web/polaris_web/web/src/apps/dashboard/pages/issues/transform.js [hunk 5]
```diff
@@ -1,82 +1,128 @@
...
+            case "IGNORED":
+                return "Ignored"
+            default:
+                return type;
+        }
+    },
+    
+    extractDomain: (url) => {
+        try {
+            const parsedUrl = new URL(url)
```
Comment (Ark2307) [inline apps/dashboard/web/polaris_web/web/src/apps/dashboard/pages/issues/transform.js:121]: use collectionsmap/hostnamemap

File: apps/dashboard/web/polaris_web/web/src/apps/dashboard/pages/observe/transform.js [hunk 6]
```diff
@@ -351,6 +351,19 @@ const transform = {
         return { sensitiveSampleData: sensitiveSampleData };
     },
 
+    getSeverityColor(severity) {
+        switch(severity.toUpperCase()) {
+            case "CRITICAL":
+                return "#E45357"
```
Comment (Ark2307) [inline apps/dashboard/web/polaris_web/web/src/apps/dashboard/pages/observe/transform.js:357]: remove this and revert to old code

File: apps/dashboard/web/polaris_web/web/src/apps/dashboard/pages/testing/TestRunResultPage/TestRunResultFlyout.jsx [hunk 7]
```diff
@@ -154,7 +144,10 @@ function TestRunResultFlyout(props) {
                         <Text color="subdued" variant="bodySm">{selectedTestRunResult?.testCategory}</Text>
                     </HorizontalStack>
                 </VerticalStack>
-                <ActionsComp />
+                <HorizontalStack gap={2} wrap={false}>
```
Comment (Ark2307) [inline apps/dashboard/web/polaris_web/web/src/apps/dashboard/pages/testing/TestRunResultPage/TestRunResultFlyout.jsx:147]: handle this correctly, use vulnerable dependency and not page wise

File: apps/dashboard/web/polaris_web/web/src/apps/dashboard/pages/testing/TestRunResultPage/TestRunResultFlyout.jsx [hunk 8]
```diff
@@ -288,7 +287,7 @@ function TestRunResultFlyout(props) {
     const tabsComponent = (
         <LayoutWithTabs
             key="tab-comp"
-            tabs={issueDetails?.id ? [overviewTab,ValuesTab]: [attemptTab]}
+            tabs={issueDetails?.id ? [overviewTab,timelineTab,ValuesTab]: [attemptTab]}
```
Comment (Ark2307) [inline apps/dashboard/web/polaris_web/web/src/apps/dashboard/pages/testing/TestRunResultPage/TestRunResultFlyout.jsx:290]: handle this on isVulnerable

File: apps/dashboard/web/polaris_web/web/src/apps/dashboard/pages/testing/TestRunResultPage/style.css [hunk 9]
```diff
@@ -1,3 +1,27 @@
...
+}
+
+#create-jira-ticket-button {
+    text-wrap: nowrap;
+}
+
+.badge-wrapper-CRITICAL .Polaris-Badge {
+    background-color: #E45357 !important;
+    color: white !important;
+}
```
Comment (Ark2307) [inline apps/dashboard/web/polaris_web/web/src/apps/dashboard/pages/testing/TestRunResultPage/style.css:12]: remove this

File: apps/dashboard/web/polaris_web/web/src/apps/dashboard/pages/testing/transform.js [hunk 10]
```diff
@@ -896,7 +896,7 @@ getRowInfo(severity, apiInfo,jiraIssueUrl, sensitiveData){
   const rowItems = [
     {
       title: 'Severity',
-      value: <Text fontWeight="semibold" color={observeFunc.getColor(severity)}>{severity}</Text>,
+      value: <Text fontWeight="semibold"><span style={{color: observeFunc.getSeverityColor(severity)}}>{severity}</span></Text>,
```
Comment (Ark2307) [inline apps/dashboard/web/polaris_web/web/src/apps/dashboard/pages/testing/transform.js:899]: remove this

File: apps/dashboard/web/polaris_web/web/src/apps/dashboard/pages/issues/IssuesPage/IssuesPage.jsx [hunk 11]
```diff
@@ -325,18 +305,144 @@ function IssuesPage(){
...
+                const key = `${item.id.testSubCategory}|${item.severity}|${item.unread.toString()}`
+                if (!uniqueIssuesMap.has(key)) {
+                    uniqueIssuesMap.set(key, {
+                        id: item.id,
+                        severity: transform.mapSeverity(item.severity),
+                        severityType: item.severity,
+                        issueName: item.id.testSubCategory,
+                        category: item.id.testSubCategory,
+                        numberOfEndpoints: 1,
+                        creationTime: item.creationTime,
```
Comment (Ark2307) [inline apps/dashboard/web/polaris_web/web/src/apps/dashboard/pages/issues/IssuesPage/IssuesPage.jsx:354]: add try catch and use null checks and undefined use `?.`

File: apps/dashboard/web/polaris_web/web/src/apps/dashboard/pages/issues/IssuesPage/IssuesPage.jsx [hunk 12]
```diff
@@ -325,18 +305,144 @@ function IssuesPage(){
...
+
+        return {value: ret, total: total}
+    }
+
+    const components = (
+        <>
+            <SummaryInfo
+                startTimestamp={startTimestamp}
+                endTimestamp={endTimestamp}
+            />
```
Comment (Ark2307) [inline apps/dashboard/web/polaris_web/web/src/apps/dashboard/pages/issues/IssuesPage/IssuesPage.jsx:399]: for every component give unique key, dont add in table, add in others

File: apps/dashboard/web/polaris_web/web/src/apps/dashboard/pages/issues/IssuesPage/IssuesPage.jsx [hunk 13]
```diff
@@ -351,25 +457,14 @@ function IssuesPage(){
...
-                    getNextUrl={getNextUrl}
-                    getStatus={func.getTestResultStatus}
-                    filterStateUrl={"/dashboard/issues"}
-                />
+            : components
             ]}
-            primaryAction={<Button primary onClick={() => openVulnerabilityReport()} disabled={showEmptyScreen}>Export vulnerability report</Button>}
+            primaryAction={
+                <HorizontalStack gap={2}>
+                    <DateRangeFilter initialDispatch={currDateRange} dispatch={(dateObj) => dispatchCurrDateRange({ type: "update", period: dateObj.period, title: dateObj.title, alias: dateObj.alias })} />
```
Comment (Ark2307) [inline apps/dashboard/web/polaris_web/web/src/apps/dashboard/pages/issues/IssuesPage/IssuesPage.jsx:464]: secondary action{date range filter}, dont add in primary, primary is export only

File: apps/dashboard/web/polaris_web/web/src/apps/dashboard/pages/issues/transform.js (hunk 5 above)
Comment (TangoBeeAkto) [reply apps/dashboard/web/polaris_web/web/src/apps/dashboard/pages/issues/transform.js:121]: using `observeFunc.getHostName()` common method

**Review by avneesh-akto — APPROVED** [review]

**Pull request merged**